app/javascript/packs/hideBookmarkButtons.js

```diff
@@ -0,0 +1,10 @@
+/* global userData */
```

Comment:
I am handling the check for the current_user's id with Javascript, since caching prevents access to the `current_user` object in Rails views.

At first, I had the Bookmark buttons hidden by default. However, this posed problems for signed-out users that were more convoluted to fix. So I decided to show the button by default, and hide it only when the post belongs to the current user.

Bear in mind that, in some instances, the button is visible for a millisecond, before it's hidden 😩 . I'm open to ways to prevent this.

Reply:
The change @Zhao-Andy describes is how we approach e.g. the Spaces roles/authorization show/hide behaviour. There are pros and cons to either approach...

Hiding by default - No weird flashes of content, but if there's any error with the JS then the elements may never show, plus on the whole we'd be manipulating _more_ elements given the majority of posts are not the author's own. All users would see save buttons 'popping' into view on all feed cards.

Showing by default, then hiding - works from the get-go even if JS is slow to load, manipulates least number of elements, but gives weird flashes of content in the case where a user is looking at their own post. Users don't see any UI changes for majority of feed cards (with the exception of on their own profile page in which case _all_ feed cards flash the button)

My 2 cents in this case is that showing by default has the least unwanted impact for users - it's probably not likely that an author is going to find many of their posts in a given page on the home/tags feed, so it's really only on the user's profile that this 'flash' feels weird to me. Definitely keen for product/design thoughts on it though as it's very noticeable when you look at your own profile page.

Reply:
(blocking) on this line in particular - I would instead use:
 
```js
import { getUserDataAndCsrfToken } from '@utilities/getUserDataAndCsrfToken';
```

which is an async helper to get userData in the JS packs. The benefit of using that helper instead of the global `userData` function is that if the userData has not yet been initialized, then it will wait until it's ready. We've had strange edge case errors in the past with using the non-async `userData()` so best to favour the more robust option we have!